extensions/ql-vscode/src/query-testing/test-manager.ts

```diff
@@ -275,68 +275,66 @@ export class TestManager extends TestManagerBase {
         // Pass the test path from the event through `Uri` and back via `fsPath` so that it matches
         // the canonicalization of the URI that we used to create the `TestItem`.
         const testItem = testsToRun.get(Uri.file(event.test).fsPath);
-        if (testItem === undefined) {
-          throw new Error(
-            `Unexpected result from unknown test '${event.test}'.`,
-          );
-        }
-
-        const duration = event.compilationMs + event.evaluationMs;
-        if (event.pass) {
-          testRun.passed(testItem, duration);
-        } else {
-          // Construct a list of `TestMessage`s to report for the failure.
-          const testMessages: TestMessage[] = [];
-          if (event.failureDescription !== undefined) {
-            testMessages.push(new TestMessage(event.failureDescription));
-          }
-          if (event.diff?.length && event.actual !== undefined) {
-            // Actual results differ from expected results. Read both sets of results and create a
-            // diff to put in the message.
-            const expected = await tryReadFileContents(
-              event.expected,
-              testMessages,
-            );
-            const actual = await tryReadFileContents(
-              event.actual,
-              testMessages,
-            );
-            if (expected !== undefined && actual !== undefined) {
-              testMessages.push(
-                TestMessage.diff(
-                  "Actual output differs from expected",
-                  expected,
-                  actual,
-                ),
-              );
+        // If the user manually adds the `--check-databases` option, we can get events for the
+        // corresponding database check tests. Just ignore these for now.
+        if (testItem !== undefined) {
```

Comment:
I think this is fine, but maybe can you add a log message for the unexpected `event.test`?